File: test_ws.py
import asyncio
import websockets
import json
import ssl

async def test():
    uri = "wss://fstream.binance.com/stream?streams=btcusdt@aggTrade/btcusdt@depth20@100ms"
    ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
    ssl_context.check_hostname = False
    ssl_context.verify_mode = ssl.CERT_NONE
    
    # We can't easily use socks5 with websockets in python without extra libs.
    # Let's try Node.js with https-proxy-agent but ignoring SSL errors.

File: test_ws.js
const WebSocket = require('ws');
const { SocksProxyAgent } = require('socks-proxy-agent');
const agent = new SocksProxyAgent('socks5://127.0.0.1:7890');
const ws = new WebSocket('wss://fstream.binance.com/stream?streams=btcusdt@aggTrade/btcusdt@depth20@100ms', { agent });
ws.on('open', () => console.log('Connected'));
ws.on('message', data => {
  const msg = JSON.parse(data);
  console.log("STREAM:", msg.stream);
  if (msg.stream.endsWith('@aggTrade')) {
      console.log('AggTrade Price:', msg.data.p);
  }
});
setTimeout(() => {
    ws.close();
    process.exit(0);
}, 3000);
